vm.go

```diff
@@ -433,8 +463,40 @@ func (session *ioSession) WaitForShim() error {
 	select {
 	case <-session.shimConnected:
 	case <-time.After(waitForShimTimeout):
-		return fmt.Errorf("timeout waiting for shim with token %s", session.token)
+		msg := fmt.Sprintf("timeout waiting for shim with token %s", session.token)
+		session.vm.infof(1, "session", msg)
+		// No need to call session.ResetShim() here. We time out because we haven't
+		// seen a shim. This error will be reported to the runtime.
+		return errors.New(msg)
 	}
+
+	return nil
+}
+
+var waitForExecTimeout = 30 * time.Second
```

Comment:
Can't make this one const, I override the timeout in tests. But sure will rename it.